modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -32,41 +34,36 @@ import weaver.scalacheck.Checkers
 
 object ConstructionServiceSuite extends MutableIOSuite with Checkers with TransactionGenerator {
 
-  type Res = (SecurityProvider[IO], KryoSerializer[F])
+  type Res = (SecurityProvider[IO], KryoSerializer[IO])
 
   override def sharedResource: Resource[IO, Res] =
     KryoSerializer
       .forAsync[IO](sharedKryoRegistrar)
       .flatMap(kryo => SecurityProvider.forAsync[IO].map(securityProvider => (securityProvider, kryo)))
 
-  def generateTestTransactions(wantSignedTransaction: Boolean)(implicit S: SecurityProvider[IO], K: KryoSerializer[IO]) = {
-    val getSignedTransaction = (transaction: Hashed[Transaction]) => transaction.signed
-    val getUnsignedTransaction = (transaction: Hashed[Transaction]) => transaction.signed.value
-
-    def txs(transactionConverter: Hashed[Transaction] => AnyRef) =
-      (KeyPairGenerator.makeKeyPair[IO], KeyPairGenerator.makeKeyPair[IO]).tupled.flatMap {
-        case (srcKey, dstKey) =>
-          val srcAddress = srcKey.getPublic.toAddress
-          val dstAddress = dstKey.getPublic.toAddress
-          val txCount = PosInt(100)
-
-          generateTransactions(srcAddress, srcKey, dstAddress, txCount)
-            .flatMap(_.traverse { hashedTransaction =>
-              KryoSerializer[F]
-                .serialize(transactionConverter(hashedTransaction))
-                .map(Hex.fromBytes(_))
-                .map((_, hashedTransaction.hash, hashedTransaction))
-                .liftTo[IO]
-            })
-      }
-
-    if (wantSignedTransaction) {
-      txs(getSignedTransaction)
-    } else {
-      txs(getUnsignedTransaction)
-    }
+  private def getBytes[G[_]: Sync](hashedTransaction: Hashed[schema.transaction.DAGTransaction], wantSignedTransaction: Boolean) = if (
```

Comment:
what's the purpose of using tagless algebra here? also, import `schema.transaction.DAGTransaction` should be fixed

Reply:
imo it's unnecessary overhead. this serialization is just a pure function, so it can be just a simple function, nothing extraordinary. also, it's overhead to use tagless algebra in tests, where everything is using IO here.
imo, there is no point for using for comprehension for a single flat map, but that's up to you.

```scala
generateTransactions(srcAddress, srcKey, dstAddress, txCount).flatMap {
  _.traverse(hashedTransaction =>
    val bytes = getBytes(hashedTransaction, wantSignedTransaction)
            
    (Hex.fromBytes(bytes), hashedTransaction.hash, hashedTransaction)
}
```

with for comp, it looks like that:

```scala
for {
  transactions <- generateTransactions(srcAddress, srcKey, dstAddress, txCount)
  testValues <- transactions.traverse { hashedTransaction =>
    val bytes = getBytes(hashedTransaction, wantSignedTransaction)
    
    (Hex.fromBytes(bytes), hashedTransaction.hash, hashedTransaction)
  }
} yield testValues
```

Reply:
Fixed the import.

Reply:
The serializer was made safe in PR https://github.com/Constellation-Labs/tessellation/pull/607

Reply:
Yep. Are you going to rebase against that branch?

Reply:
Removed the tagless algebra from the tests.

Reply:
@marcinwadon , I was going to wait until that branch gets merged to develop.